mypy.ini

```diff
@@ -0,0 +1,9 @@
+[mypy]
+disallow_untyped_defs = True
+exclude = ^((tests|docs|examples|bin)/|conftest.py|src/clib/|src/xtgeo/(common|cube|xtgeo|grid3d|metadata|plot|roxutils|surface|well|xyz)/?)
+follow_imports = skip
+ignore_missing_imports = True
+python_version = 3.8
+
+[mypy-xtgeo.grid3d.*]
```

Comment:
Been on an insane journey here... But i was not aware that setting `ignore_missing_imports = True` also means that things imported into your module will also be ignored. In the prints below, i look at `reveal_type(...)` from `_optional_index_lookup(...)` The first without [mypy-xtgeo.grid3d.*] the second one with. The second one is right, as it has the expected types, and not `Any`.

```bash
(xgp11) ➜  xtgeo git:(add-types-ecl_inte_head) ✗ mypy src/xtgeo/grid3d/_ecl_inte_head.py | grep src/xtgeo/grid3d/_ecl_inte_head.py
src/xtgeo/grid3d/_ecl_inte_head.py:80: note: Revealed type is "Union[Any, None]"
src/xtgeo/grid3d/_ecl_inte_head.py:91: note: Revealed type is "Union[builtins.int, None]"
(xgp11) ➜  xtgeo git:(add-types-ecl_inte_head) ✗ mypy src/xtgeo/grid3d/_ecl_inte_head.py | grep src/xtgeo/grid3d/_ecl_inte_head.py
src/xtgeo/grid3d/_ecl_inte_head.py:80: note: Revealed type is "Union[xtgeo.grid3d._ecl_output_file.UnitSystem, None]"
src/xtgeo/grid3d/_ecl_inte_head.py:91: note: Revealed type is "Union[builtins.int, None]"
```

POC, without the updated mypy.ini we would get the below results.

```python
from enum import Enum
from ... import Units
    
class Cod(Enum):
    atlantic = 0
    pacific = 1

reveal_type(Cod(1))         # Cod as expected.
reveal_type(Units(1))    # Any as unexpected.
```